lib/plan_executor/executor.rb

```diff
@@ -0,0 +1,218 @@
+# frozen_string_literal: true
+
+# Used for $ERROR_INFO. This *must* be capitalized!
+require 'English'
+require 'json'
+require 'logging'
+require 'set'
+require 'bolt/result'
+require 'bolt/config'
+require 'bolt/transport/api'
+require 'bolt/notifier'
+require 'bolt/result_set'
+require 'bolt/puppetdb'
+
+module PlanExecutor
+  class Executor
+    attr_reader :noop, :transport
+    attr_accessor :run_as
+
+    # CODEREVIEW: This assume concurrency is configured in the orchestrator
+    def initialize(noop = nil,
+                   bundled_content: nil,
+                   load_config: true)
+      @bundled_content = bundled_content
+      @logger = Logging.logger[self]
+      @plan_logging = false
+      @load_config = load_config
+      @noop = noop
+      @run_as = nil
+      @logger.debug { "Started" }
+      @notifier = Bolt::Notifier.new
+      @transport = Bolt::Transport::Api.new
+    end
+
+    # This handles running the job, catching errors, and turning the result
+    # into a result set
+    def execute(targets)
+      result_array = begin
+                       results = yield
+                     rescue StandardError => e
+                       @logger.warn(e)
+                       # CODEREVIEW how should we fail if there's an error?
+                       Array(Bolt::Result.from_exception(targets[0], e))
+                     end
+      Bolt::ResultSet.new(result_array)
+    end
+
+    def log_action(description, targets)
```

Comment:
these log levels are probably wrong for a service. I'm tempted to remove it and add back in service specific logging later. If it starts to cause problems its' probably worth just removing.

Reply:
In general "plan logging" should not be part of the logger but about sending log events to orchestrator in a plan service. run_* events are already handled by the task command already so removing this would only effect "run_plan"